configserver/src/main/java/com/yahoo/vespa/config/server/application/Application.java

```diff
@@ -136,14 +138,21 @@ public ConfigResponse resolveConfig(GetConfigRequest req, ConfigResponseFactory
 
         ConfigResponse configResponse = responseFactory.createResponse(payload, def.getCNode(), appGeneration, internalRedeploy);
         metricUpdater.incrementProcTime(System.currentTimeMillis() - start);
-        if ( ! req.noCache()) {
+        if (useCache(req)) {
             cache.put(cacheKey, configResponse, configResponse.getConfigMd5());
             metricUpdater.setCacheConfigElems(cache.configElems());
             metricUpdater.setCacheChecksumElems(cache.checkSumElems());
         }
         return configResponse;
     }
 
+    private boolean useCache(GetConfigRequest request) {
+        if (request.noCache())
+            return false;
+        else
+            return new FeatureFlag("use-config-server-cache", true, new FileFlagSource()).value();
```

Comment:
This is how I would do it: The FileFlagSource should get injected. Define the FeatureFlag only once in the constructor, taking in the source. You would have a `private final FeatureFlag useConfigServerCache` that you'd call value() on here.